libraries/AP_Motors/AP_Motors_Thrust_Linearization.cpp

```diff
@@ -24,6 +74,8 @@ Thrust_Linearization::Thrust_Linearization(AP_Motors& _motors) :
     // setup battery voltage filtering
     batt_voltage_filt.set_cutoff_frequency(AP_MOTORS_BATT_VOLT_FILT_HZ);
     batt_voltage_filt.reset(1.0);
+
+    AP_Param::setup_object_defaults(this, var_info);
```

Comment:
I think we need to define this out on multi copters and plane. Otherwise we will get the heli defaults from the param table rather than the copter ones.

Eventuality we can do the param conversion and go back to normal.

Reply:
Sorry, no. 

Sorry for explaining badly, to elaborate on the problem slightly more. The issue is we load defaults to the same variables in the `AP_MotorsMulticopter` class here:

https://github.com/ArduPilot/ardupilot/blob/514434193d7c4dba259180db4a9aa83391c7cf01/libraries/AP_Motors/AP_MotorsMulticopter.cpp#L47-L75

And the rest further down. Then the same variables are initialised a second time when this object is created. So we have the variables with two defaults. This is because multirotor is not using the new param table but rather including the variables individually in the higher level table. 

So the fix is to either move to the new table on multirotor too, but then we need to do the param conversion. Or to remove the new table on multirotor. 

So you should be able to just do this:
```suggestion
#if APM_BUILD_TYPE(APM_BUILD_Heli)
    AP_Param::setup_object_defaults(this, var_info);
 #endif
```

You can check it worked by looking at the size change on copter, currently this adds 148 bytes. You can check in the build size action: 
https://github.com/ArduPilot/ardupilot/actions/runs/5242625088/jobs/9466260671?pr=22661

If copter is not using the table then the size change should be 0. 

Ultimately the longer term fix would be to move copter over to the new param table, I can look at adding that to this PR or as a future PR so we can remove this work around. There may also be a issue for quadplane with the depth of parameter trees that this avoids because heli is not supported on quadplane. Here its "copter -> motors -> thrust lin" on plane it would be "plane -> quadpalne -> motors -> thrust lin" that one level deeper might break params which would be one reason we could not use the new param table on copter with out a larger parameter tree depth fix.